spec/cordova/platform/addHelper.spec.js

```diff
@@ -16,34 +16,439 @@
+        spyOn(shell, 'mkdir');
+        spyOn(fs, 'existsSync').and.returnValue(false);
+        spyOn(fs, 'writeFileSync');
+        spyOn(cordova_util, 'projectConfig').and.returnValue(path.join(projectRoot, 'config.xml'));
+        spyOn(cordova_util, 'isDirectory').and.returnValue(false);
+        spyOn(cordova_util, 'fixRelativePath').and.callFake(function (input) { return input; });
+        spyOn(cordova_util, 'isUrl').and.returnValue(false);
+        spyOn(cordova_util, 'hostSupports').and.returnValue(true);
+        spyOn(cordova_util, 'removePlatformPluginsJson');
+        spyOn(cordova_config, 'read').and.returnValue({});
+        spyOn(events, 'emit');
+        // Fake platform details we will use for our mocks, returned by either
+        // getPlatfromDetailsFromDir (in the local-directory case), or
+        // downloadPlatform (in every other case)
+        spyOn(platform_module, 'getPlatformDetailsFromDir').and.returnValue(Q(fake_platform));
+        spyOn(platform_addHelper, 'downloadPlatform').and.returnValue(Q(fake_platform));
+        spyOn(platform_addHelper, 'getVersionFromConfigFile').and.returnValue(false);
+        spyOn(platform_addHelper, 'installPluginsForNewPlatform').and.returnValue(Q());
+        platform_api_mock = jasmine.createSpyObj('platform api mock', ['createPlatform', 'updatePlatform']);
+        platform_api_mock.createPlatform.and.returnValue(Q());
+        platform_api_mock.updatePlatform.and.returnValue(Q());
+        spyOn(cordova_util, 'getPlatformApiFunction').and.returnValue(platform_api_mock);
+        spyOn(platform_metadata, 'save');
+        spyOn(cordova_util, 'requireNoCache').and.returnValue({});
+    });
+    afterEach(function () {
+        cfg_parser_revert_mock();
+        fetch_revert_mock();
+        prepare_revert_mock();
+    });
     describe('error/warning conditions', function () {
-        it('should require specifying at least one platform');
-        it('should warn if host OS does not support the specified platform');
+        it('should require specifying at least one platform', function (done) {
+            platform_addHelper('add', hooks_mock).then(function () {
+                fail('addHelper success handler unexpectedly invoked');
+            }).fail(function (e) {
+                expect(e.message).toContain('No platform specified.');
+            }).done(done);
+        });
+
+        it('should log if host OS does not support the specified platform', function (done) {
+            cordova_util.hostSupports.and.returnValue(false);
+            platform_addHelper('add', hooks_mock, projectRoot, ['atari']).then(function() {
+        fail('addHelper success handler unexpectedly invoked');
+            }).fail(function (e) {
+                expect(cordova_util.hostSupports).toHaveBeenCalled();
+                expect(events.emit).toHaveBeenCalledWith('warning', 'WARNING: Applications for platform atari can not be built on this OS - darwin.');
```

Comment:
This test will always fail somewhere ... currently failing on Travis because the OS is `linux` and not `darwin` and also failing on Appveyor where process.platform is `win32`
Isn't it enough to just test that the call was a 'warning' and started with 'WARNING: Applications for platform' ?